pennylane/templates/subroutines/swap_test.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``SWAP Test`` template.
+"""
+import pennylane as qml
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def SWAPTest(register1, register2, ancilla):
+
+    r""" A controlled-SWAP subroutine that can be used to create a
+    `SWAP test <https://en.wikipedia.org/wiki/Swap_test>`__ between
+    two registers.
+
+    Args:
+        register1 (Iterable or Wires): The first register of wires passed into the template.
+        register2 (Iterable or Wires): The second register of wires passed into the template.
+        ancilla (int or Wires): The ancilla qubit that controls the CSWAP gates in the template.
+
+    Raises:
```

Comment:
Not sure if declaring when an error is raised is standard in PL :thinking:

Reply:
The other templates have an error declaration, so I assumed that it should be done here as well

Reply:
There's no established recommendation here (e.g., I don't usually list raised errors, but I know others do)